charts/src/main/java/hu/ma/charts/line/LineChart.kt

```diff
@@ -0,0 +1,363 @@
+package hu.ma.charts.line
+
+import android.graphics.Paint
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
+import androidx.compose.foundation.gestures.detectHorizontalDragGestures
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.requiredHeight
+import androidx.compose.foundation.layout.requiredSize
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Dp
+import hu.ma.charts.legend.DrawHorizontalLegend
+import hu.ma.charts.legend.DrawVerticalLegend
+import hu.ma.charts.line.data.DrawAxis
+import hu.ma.charts.line.data.LineChartData
+import hu.ma.charts.legend.LegendPosition
+import hu.ma.charts.pie.createLegendEntries
+import kotlin.math.abs
+import kotlin.math.max
+
+@Composable
+fun LineChart(
+  chartHeight: Dp? = null,
+  data: LineChartData,
+  onDrillDown: ((x: Int, series: List<LineChartData.SeriesData>) -> Unit)? = null,
+) {
+
+  val legendEntries = remember(data) { data.createLegendEntries(data.legendShapeSize) }
+
+  @Composable
+  fun legend() {
+    when (data.legendPosition) {
+      LegendPosition.End, LegendPosition.Start -> {
+        DrawVerticalLegend(legendEntries)
+      }
+      LegendPosition.Top, LegendPosition.Bottom ->
+        DrawHorizontalLegend(legendEntries)
+    }
```

Comment:
This should be customisable, so let's pass a single lambda (`@Composable (position, entries) -> Unit`) for both positions, and have this be a default value instead

Reply:
I had some trouble understanding the customized part of the legends for pie and bars. Do you mean that instead of hardcoding horizontal and vertical legend here, we should let the user specify whether to use `DrawHorizontalLegend` or `DrawVerticalLegend` in the lambda, and have my implementation here as the default?

Reply:
Yep, you'll have some composable defined that will contain these calls and be passed in as a `legend` param  

And reasoning for this is to have the legend be non-final and customisable by the end user, but with sane defaults and certain constraints (position and items, and maybe something else 😅 )